### ISSUE TITLE
FreeBSD build fails

### ISSUE THREAD

**Issue by labdsf** [dino/dino#344]
There are multiple issues, starting with `configure` script.

Shebang in configure script is `#!/bin/bash`. On FreeBSD, bash is installed in `/usr/local/bin/bash`. It should be replaced with either `#!/usr/bin/env bash` or `#!/bin/sh`.

Then, `getopt` utility is used to parse long options. FreeBSD `getopt` does not support long options. Getopt with GNU-style options can be installed from ports, but it is installed in `/usr/local/bin/getopt` which goes after `/usr/bin/getopt` in PATH by default.

After replacing `getopt` with `/usr/local/bin/getopt`, I can run `configure --help` and such.

Then, even if I disable OMEMO plugin, it fails with `/usr/bin/ld: cannot find -lgcrypt`, so I have to add GCrypt to `plugins/omemo/CMakeLists.txt`.

Now I get error `error: Package `libgcrypt' not found in specified Vala API directories or GObject-Introspection GIR directories`, don't know how to fix it.

**Comment by labdsf** [comment]
PR #335 seems related

**Comment by labdsf** [comment]
As for unportable configure script, I am preparing a patch on top of PR #350 to remove all bashisms and `getopt`. PR #350 changes a line with bashism (`==`), so I will submit it once it is merged.

**Comment by nwhitehorn** [comment]
A patch that fixes the CMake problems is at http://people.freebsd.org/~nwhitehorn/dino.diff.

**Comment by labdsf** [comment]
@nwhitehorn Thanks, works for me. Will you submit a PR?

**Comment by bodems** [comment]
I also confirm that it builds now on FreeBSD 12. Thanks for your patch @nwhitehorn !

**Comment by Echolon** [comment]
@link2xt, @bodems This thread can be closed?

**Comment by link2xt** [comment]
@Echolon the patch hasn't been merged into this repository it seems. Do you expect freebsd users to keep patching the source?

**Comment by Echolon** [comment]
Hi link2xt,
no expectations, just asking.

**Comment by link2xt** [comment]
> A patch that fixes the CMake problems is at http://people.freebsd.org/~nwhitehorn/dino.diff.

Could you merge the linked patch first?

I don't have freebsd around anymore, so can't test, but the patch does reasonable things: replaces some hardcoded library names with variables.
